sim/src/channel.rs

```diff
@@ -62,8 +62,8 @@ impl SimElement for Channel {
         assert!(self.neighbor.len() < 2);
         self.neighbor.push(neighbor);
     }
-    fn whoami(&self) -> (&str, u32, Vec<u32>) {
-        return ("Channel", self.id, self.neighbor.clone());
+    fn whoami(&self) -> (bool, u32, Vec<u32>) {
+        return (false, self.id, self.neighbor.clone());
```

Comment:
Every rpc carries with it a record of where it has been.  However, in order to simplify the path, we only include nodes in the path right now, because if we included every channel, every plugin, it would be a bit silly, long, and hard to read.  Because we created the graph, we know that when it goes through a certain node, it also goes through the associated plugin.  And if it jumps from one node to another, clearly it's going through an edge between them.

The path's purpose is twofold.  It is useful both for the filter to understand in what order the rpc has visited all the nodes for implementing the WHERE clause.  It also makes it much easier to trace the path of a particular rpc when you're looking at the printed output.

Reply:
Where it has been means the source node? What does that imply? So this is close to the design of the traces that are used in actual deployments?

Reply:
In the real world, if a trace were to go to frontend-checkoutservice-paymentservice, this path will have all of those in a line.  It's a history of where the rpc has been.  If you look at the c++ filters, on response headers inbound, they can query a b3-spanid-path header, and the history given there is used to construct the trace graph.  The path has the same purpose.

Reply:
So are we essentially modelling trace-id and the span with this?

Reply:
I'm not sure what you mean.  Are you suggesting we scrap this and make it another field in the headers hashmap?  I think that would make sense.

Reply:
Yeah I think that is the most sensible option. If in the reference system this is done over HTTP headers we should do the same thing.